qcodes/instrument_drivers/Andor/DU401.py

```diff
@@ -0,0 +1,424 @@
+        20097: 'DRV_DIVIDE_BY_ZERO_ERROR', 20098: 'DRV_INVALID_RINGEXPOSURES', 20990: 'DRV_ERROR_NOCAMERA',
+        20991: 'DRV_NOT_SUPPORTED', 20992: 'DRV_NOT_AVAILABLE', 20115: 'DRV_ERROR_MAP', 20116: 'DRV_ERROR_UNMAP',
+        20117: 'DRV_ERROR_MDL', 20118: 'DRV_ERROR_UNMDL', 20119: 'DRV_ERROR_BUFFSIZE', 20121: 'DRV_ERROR_NOHANDLE',
+        20130: 'DRV_GATING_NOT_AVAILABLE', 20131: 'DRV_FPGA_VOLTAGE_ERROR', 20099: 'DRV_BINNING_ERROR',
+        20100: 'DRV_INVALID_AMPLIFIER', 20101: 'DRV_INVALID_COUNTCONVERT_MODE'}
+
+    def __init__(self, dll_path=None, verbose=False):
+
+        self.verbose = verbose
+        self.dll = ctypes.windll.LoadLibrary(dll_path or self._dll_path)
+
+    def error_check(self, code, function_name=''):
+        if code in self._success_codes.keys():
+            if self.verbose:
+                print("atmcd64d: [%s]: %s" % (function_name, self._success_codes[code]))
+        elif code in self._error_codes.keys():
+            print("atmcd64d: [%s]: %s" % (function_name, self._error_codes[code]))
+            raise Exception(self._error_codes[code])
+        else:
+            print("atmcd64d: [%s]: Unknown code: %s" % (function_name, code))
+            raise Exception()
+
+    def cooler_off(self):
+        code = self.dll.CoolerOFF()
+        self.error_check(code, 'CoolerOFF')
+
+    def cooler_on(self):
+        code = self.dll.CoolerON()
+        self.error_check(code, 'CoolerON')
+
+    def get_acquired_data(self, size):
+        c_data_array = ctypes.c_int * size
+        c_data = c_data_array()
+        code = self.dll.GetAcquiredData(ctypes.pointer(c_data), size)
+        self.error_check(code, 'GetAcquiredData')
+        acquired_data = []
+        for i in range(len(c_data)):
+            acquired_data.append(c_data[i])
+        return acquired_data
+
+    def get_acquisition_timings(self):
+        c_exposure = ctypes.c_float()
+        c_accumulate = ctypes.c_float()
+        c_kinetic = ctypes.c_float()
+        code = self.dll.GetAcquisitionTimings(ctypes.byref(c_exposure), ctypes.byref(c_accumulate), ctypes.byref(c_kinetic))
+        self.error_check(code, 'GetAcquisitionTimings')
+        return c_exposure.value, c_accumulate.value, c_kinetic.value
+
+    def get_camera_handle(self, camera_index):
+        c_camera_handle = ctypes.c_long()
+        code = self.dll.GetCameraHandle(camera_index, ctypes.byref(c_camera_handle))
+        self.error_check(code, 'GetCameraHandle')
+        return c_camera_handle.value
+
+    def get_camera_serial_number(self):
+        c_serial_number = ctypes.c_int()
+        code = self.dll.GetCameraSerialNumber(ctypes.byref(c_serial_number))
+        self.error_check(code, 'GetCameraSerialNumber')
+        return c_serial_number.value
+
+    def get_hardware_version(self):
+        c_pcb = ctypes.c_int()
+        c_decode = ctypes.c_int()
+        c_dummy1 = ctypes.c_int()
```

Comment:
the dummy arguments, do they not have any meaning? or smth like "reserved" by the API? Anyway, please add a comment for this in the code (or rename them to something more clear if possible).